src/autocomplete/dataset.js

```diff
@@ -37,6 +37,8 @@ function Dataset(o) {
 
   this.debounce = o.debounce;
 
+  this.disableCache = !!o.disableCache;
```

Comment:
I would go for `enableCache` (or something similar) with a default value to `true` rather than `disableCache`. Positive condition are a bit easier to reason about.